x-pack/plugins/cloud_security_posture/server/tasks/findings_stats_task.ts

```diff
@@ -130,31 +190,45 @@ const aggregateLatestFindings = async (
       ).toFixed(2)}ms]`
     );
 
-    const clustersStats = Object.fromEntries(
-      evaluationsQueryResult.aggregations.score_by_cluster_id.buckets.map(
-        (clusterStats: AggregatedFindingsByCluster) => {
+    // getting score per policy template buckets
+    const scoresByPolicyTemplatesBuckets =
+      scoreIndexQueryResult.aggregations.score_by_policy_template.buckets;
+
+    // iterating over the buckets and return promises which will index a modified document into the scores index
+    const docIndexingPromises = scoresByPolicyTemplatesBuckets.map(async (policyTemplateTrend) => {
+      // creating score per cluster id objects
+      const clustersStats = Object.fromEntries(
+        policyTemplateTrend.score_by_cluster_id.buckets.map((clusterStats) => {
+          const clusterId = clusterStats.key;
+
           return [
-            clusterStats.key,
+            clusterId,
             {
               total_findings: clusterStats.total_findings.value,
               passed_findings: clusterStats.passed_findings.doc_count,
               failed_findings: clusterStats.failed_findings.doc_count,
             },
           ];
-        }
-      )
-    );
+        })
+      );
+
+      // each document contains the policy template and its scores
+      return await esClient.index({
```

Comment:
we can remove this `await` and the `async` as `index` returns a promise, `docIndexingPromises` type would be the same